src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -724,6 +724,16 @@ private void WriteFrameworkAssemblies()
                     return;
                 }
 
+                //  Keep track of Framework assemblies that we've already written items for,
+                //  in order to only create one item for each Framework assembly.
+                //  This means that if multiple packages have a dependency on the same
+                //  Framework assembly, we will no longer emit separate items for each one.
+                //  This should make the logs a lot cleaner and easier to understand,
+                //  and may improve perf.  If you really want to know all the packages
+                //  that brought in a framework assembly, you can look in the assets
+                //  file.
+                var writtenFrameworkAssemblies = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
Nit you still may have duplicates if folks do stuff like mscorLib or system.core and System.Core.  I have seen this sort of thing in packages, will anything break?

Reply:
No it shouldn't break.  In a lot of cases you will still have 2 copies of a framework reference, one from your project file or implicitly from the SDK targets, and one from the framework assemblies from NuGet packages.  But 2 is a lot better in the logs than having 5 or more because several different NuGet packages had the same framework assembly dependency.